source/vibe/http/common.d

```diff
@@ -469,20 +488,25 @@ final class ChunkedOutputStream : OutputStream {
 	{
 		assert(!m_finalized);
 		if( m_buffer.data.length > 0 ) flush();
+		ubyte[] buffer = null;
+		auto getBytes = (size_t size) {
+			if (buffer is null || buffer.length < size)
+				buffer = new ubyte[](size);
+				return buffer[0..size];
+			};
+
 		if( nbytes == 0 ){
 			while( !data.empty ){
 				auto sz = data.leastSize;
 				assert(sz > 0);
-				writeChunkSize(sz);
-				m_out.write(data, sz);
-				m_out.write("\r\n");
-				m_out.flush();
+				auto bytes = getBytes(sz);
+				data.read(bytes);
+				write(bytes);
```

Comment:
Flushing can indeed be removed here, it was just necessary to cleanly switch between buffered and stream modes. I just realized now that `m_buffer` is an `AllocAppender`. We'd need something like `AllocAppender.put(scope size_t delegate(scope T[] dst) del);`, where the delegate is called with `m_remaining` and the return value is the number of bytes written to that buffer.